


![](img/add_cell.png)



To make a new cell you can click on the **+** in the toolbar above.

To change the cell to a text cell, you can click on the ***Code*** dropdown menu above and change it to ***Markdown***.

Delete this cell (by right clicking and selecting ***Delete Cells***) before handing in your project. 

# Data Analysis for the Disney Movies

# Introduction : 

In this analysis there will be analysis of various question associated with the collection of Disney movies.I am interested in finding out regarding the Disney movies dataset.

## Questions of Interests:

In this analysis I will be investigating a series of questions based on the Disney dataset. I am interested in finding out which disney movie was the highest total gross revenue and which genre had highest revenue based on the genre category. It would also be interested to compare how eachcategory has performed including the benchmark and excluding the benchmark to get an analysis whether there is any change on ranking on genre types. 

# Methods and Results

I am  interested in computing the Disney Movies collection on the basis of the revenue as I will be conducting my analysis. This implies that I will only be using only **disney movies total gross** tables as that is only tables which has a relevant data for analysis.

In [9]:
import altair as alt
import pandas as pd
import numpy as np

In [10]:
disney_movies = pd.read_csv('disney_movies_total_gross.csv')
disney_movies

movie_title  release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                          Pinocchio   Feb 9, 1940  Adventure           G   
2                           Fantasia  Nov 13, 1940    Musical           G   
3                  Song of the South  Nov 12, 1946  Adventure           G   
4                         Cinderella  Feb 15, 1950      Drama           G   
..                               ...           ...        ...         ...   
574         The Light Between Oceans   Sep 2, 2016      Drama       PG-13   
575                   Queen of Katwe  Sep 23, 2016      Drama          PG   
576                   Doctor Strange   Nov 4, 2016  Adventure       PG-13   
577                            Moana  Nov 23, 2016  Adventure          PG   
578     Rogue One: A Star Wars Story  Dec 16, 2016  Adventure       PG-13   

      total_gross inflation_adjusted_gross  
0    $184,925,485           $5,228,953,251  
1     $84,300,000           $2,188,229,052  
2     $83,320,000           $2,187,090,808  
3     $65,000,000           $1,078,510,579  
4     $85,000,000             $920,608,730  
..            ...                      ...  
574   $12,545,979              $12,545,979  
575    $8,874,389               $8,874,389  
576  $232,532,923             $232,532,923  
577  $246,082,029             $246,082,029  
578  $529,483,936             $529,483,936  

[579 rows x 6 columns]

Let's import the libraries and the table to load the data

In [11]:
print(disney_movies.head(579))
for columns in disney_movies.total_gross:
    disney_movies['total_gross'] = disney_movies['total_gross'].str.replace(r'\W',"",regex=True)

                         movie_title  release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                          Pinocchio   Feb 9, 1940  Adventure           G   
2                           Fantasia  Nov 13, 1940    Musical           G   
3                  Song of the South  Nov 12, 1946  Adventure           G   
4                         Cinderella  Feb 15, 1950      Drama           G   
..                               ...           ...        ...         ...   
574         The Light Between Oceans   Sep 2, 2016      Drama       PG-13   
575                   Queen of Katwe  Sep 23, 2016      Drama          PG   
576                   Doctor Strange   Nov 4, 2016  Adventure       PG-13   
577                            Moana  Nov 23, 2016  Adventure          PG   
578     Rogue One: A Star Wars Story  Dec 16, 2016  Adventure       PG-13   

      total_gross inflation_adjusted_gross  
0    $184,925,485           $5

In [12]:
print(disney_movies.head(579))
for columns in disney_movies.inflation_adjusted_gross:
    disney_movies['inflation_adjusted_gross'] = disney_movies['inflation_adjusted_gross'].str.replace(r'\W',"",regex=True)

                         movie_title  release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                          Pinocchio   Feb 9, 1940  Adventure           G   
2                           Fantasia  Nov 13, 1940    Musical           G   
3                  Song of the South  Nov 12, 1946  Adventure           G   
4                         Cinderella  Feb 15, 1950      Drama           G   
..                               ...           ...        ...         ...   
574         The Light Between Oceans   Sep 2, 2016      Drama       PG-13   
575                   Queen of Katwe  Sep 23, 2016      Drama          PG   
576                   Doctor Strange   Nov 4, 2016  Adventure       PG-13   
577                            Moana  Nov 23, 2016  Adventure          PG   
578     Rogue One: A Star Wars Story  Dec 16, 2016  Adventure       PG-13   

    total_gross inflation_adjusted_gross  
0     184925485           $5,228

As the dataset above is not cleaned and had dollar($) and comma(,) symbol inserted in the total_gross and inflation_adjusted_gross and had to remove those values which will help to sort data in ascending and the descending order to evaluate which movie has highest totalgross and lowest totalgross respectively based on the condition.

In [13]:
disney_movies.dtypes

movie_title                 object
release_date                object
genre                       object
MPAA_rating                 object
total_gross                 object
inflation_adjusted_gross    object
dtype: object

In [14]:
disney_movies['release_date'] = pd.to_datetime(disney_movies.release_date)
disney_movies

movie_title release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                          Pinocchio   1940-02-09  Adventure           G   
2                           Fantasia   1940-11-13    Musical           G   
3                  Song of the South   1946-11-12  Adventure           G   
4                         Cinderella   1950-02-15      Drama           G   
..                               ...          ...        ...         ...   
574         The Light Between Oceans   2016-09-02      Drama       PG-13   
575                   Queen of Katwe   2016-09-23      Drama          PG   
576                   Doctor Strange   2016-11-04  Adventure       PG-13   
577                            Moana   2016-11-23  Adventure          PG   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

    total_gross inflation_adjusted_gross  
0     184925485               5228953251  
1      84300000               2188229052  
2      83320000               2187090808  
3      65000000               1078510579  
4      85000000                920608730  
..          ...                      ...  
574    12545979                 12545979  
575     8874389                  8874389  
576   232532923                232532923  
577   246082029                246082029  
578   529483936                529483936  

[579 rows x 6 columns]

In [31]:
disney_movies['total_gross'] = disney_movies['total_gross'].astype(np.int64)
disney_movies['inflation_adjusted_gross'] = disney_movies['inflation_adjusted_gross'].astype(np.int64)
disney_movies

movie_title release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                          Pinocchio   1940-02-09  Adventure           G   
2                           Fantasia   1940-11-13    Musical           G   
3                  Song of the South   1946-11-12  Adventure           G   
4                         Cinderella   1950-02-15      Drama           G   
..                               ...          ...        ...         ...   
574         The Light Between Oceans   2016-09-02      Drama       PG-13   
575                   Queen of Katwe   2016-09-23      Drama          PG   
576                   Doctor Strange   2016-11-04  Adventure       PG-13   
577                            Moana   2016-11-23  Adventure          PG   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

     total_gross  inflation_adjusted_gross  
0      184925485                5228953251  
1       84300000                2188229052  
2       83320000                2187090808  
3       65000000                1078510579  
4       85000000                 920608730  
..           ...                       ...  
574     12545979                  12545979  
575      8874389                   8874389  
576    232532923                 232532923  
577    246082029                 246082029  
578    529483936                 529483936  

[579 rows x 6 columns]

In [32]:
disney_movies.dtypes

movie_title                         object
release_date                datetime64[ns]
genre                               object
MPAA_rating                         object
total_gross                          int64
inflation_adjusted_gross             int64
dtype: object

The release_date,total_gross and inflation_adjusted_gross is not in a proper dtypes so conveted in a proper dtypes which will help to sort data based on the given condition

In [33]:
disney00 = disney_movies.drop(columns = ['MPAA_rating'])
disney00

movie_title release_date      genre  total_gross  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical    184925485   
1                          Pinocchio   1940-02-09  Adventure     84300000   
2                           Fantasia   1940-11-13    Musical     83320000   
3                  Song of the South   1946-11-12  Adventure     65000000   
4                         Cinderella   1950-02-15      Drama     85000000   
..                               ...          ...        ...          ...   
574         The Light Between Oceans   2016-09-02      Drama     12545979   
575                   Queen of Katwe   2016-09-23      Drama      8874389   
576                   Doctor Strange   2016-11-04  Adventure    232532923   
577                            Moana   2016-11-23  Adventure    246082029   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure    529483936   

     inflation_adjusted_gross  
0                  5228953251  
1                  2188229052  
2                  2187090808  
3                  1078510579  
4                   920608730  
..                        ...  
574                  12545979  
575                   8874389  
576                 232532923  
577                 246082029  
578                 529483936  

[579 rows x 5 columns]

In [34]:
disney_df = disney00.sort_values('total_gross',ascending = False)
disney_df

movie_title release_date      genre  \
564    Star Wars Ep. VII: The Force Awakens   2015-12-18  Adventure   
524                            The Avengers   2012-05-04     Action   
578            Rogue One: A Star Wars Story   2016-12-16  Adventure   
571                            Finding Dory   2016-06-17  Adventure   
558                 Avengers: Age of Ultron   2015-05-01     Action   
..                                      ...          ...        ...   
511                                Zokkomon   2011-04-22  Adventure   
20   The Many Adventures of Winnie the Pooh   1977-03-11        NaN   
29                                Condorman   1981-08-07     Action   
355                     Frank McKlusky C.I.   2002-01-01        NaN   
27                                      Amy   1981-03-20      Drama   

     total_gross  inflation_adjusted_gross  
564    936662225                 936662225  
524    623279547                 660081224  
578    529483936                 529483936  
571    486295561                 486295561  
558    459005868                 459005868  
..           ...                       ...  
511         2815                      2984  
20             0                         0  
29             0                         0  
355            0                         0  
27             0                         0  

[579 rows x 5 columns]

Now that we have cleaned and have sorted the data based on the highest total_gross revenue earned by each Disney film and as it was obvious that Star Wars was the highest earning movie as it was the most popular Disney franchise and the  MPAA_Rating was not necessary for the analysis so I have deleted it.  

In [35]:
disney = disney_df['total_gross'].astype('int').sum()
disney

37462335553

As the data was already cleaned above we need to find out what was the total gross revenue was earned by disney movies from  1937 until 2016 during this period.

In [36]:
disney0= disney_df['total_gross'].astype('int').mean()
disney0

64701788.51986183

We need to find out the mean for disney movies to know which movies has performed better and vice-versa and which genre category has performed well in our further analysis  

In [37]:
disney1 = disney_df[(disney_df['total_gross']>=disney0)]
disney1

movie_title release_date                genre  \
564    Star Wars Ep. VII: The Force Awakens   2015-12-18            Adventure   
524                            The Avengers   2012-05-04               Action   
578            Rogue One: A Star Wars Story   2016-12-16            Adventure   
571                            Finding Dory   2016-06-17            Adventure   
558                 Avengers: Age of Ultron   2015-05-01               Action   
..                                      ...          ...                  ...   
57                                 Stakeout   1987-08-05               Comedy   
318              Deuce Bigalow: Male Gigolo   1999-12-10               Comedy   
442                                 Step Up   2006-08-11                Drama   
464  Hannah Montana/Miley Cyrus: Best of B…   2008-02-01  Concert/Performance   
3                         Song of the South   1946-11-12            Adventure   

     total_gross  inflation_adjusted_gross  
564    936662225                 936662225  
524    623279547                 660081224  
578    529483936                 529483936  
571    486295561                 486295561  
558    459005868                 459005868  
..           ...                       ...  
57      65673233                 141592160  
318     65535067                 106921374  
442     65328121                  84078780  
464     65281781                  76646993  
3       65000000                1078510579  

[168 rows x 5 columns]

As we found out that 168  movies had performed better than the average and remainning all the movies were below average based on the mean value which we had calculated earlier. 

In [38]:
disney2 = disney_df['genre']
disney3 = disney2.value_counts()
disney3

Comedy                 182
Adventure              129
Drama                  114
Action                  40
Thriller/Suspense       24
Romantic Comedy         23
Musical                 16
Documentary             16
Western                  7
Horror                   6
Black Comedy             3
Concert/Performance      2
Name: genre, dtype: int64

In [39]:
disney4 = disney1['genre']
disney5 = disney4.value_counts() 
disney5

Adventure              65
Comedy                 44
Drama                  20
Action                 17
Thriller/Suspense       7
Musical                 6
Romantic Comedy         6
Western                 2
Concert/Performance     1
Name: genre, dtype: int64

As the above 2 tables reflect data about various genre types and how many movies are there in each type in a dataset. The 1st table reflects all the movies category which has performed below benchmark and also includes those movies which has performed better than the average and in the second table it includes only those genre movies which has performed better than the average.

In [40]:
disney6 = pd.DataFrame(disney_df.groupby('genre')['total_gross'].mean().sort_values(ascending=True)).astype('int64')
disney6 = disney6.reset_index()
disney6

genre  total_gross
0           Documentary     11292851
1                Horror     14511478
2          Black Comedy     32514404
3                 Drama     36026078
4                Comedy     44613294
5       Romantic Comedy     50095950
6               Western     51287351
7   Concert/Performance     51728233
8     Thriller/Suspense     58616938
9               Musical     72330259
10               Action    104614082
11            Adventure    127047050

In [41]:
disney7 = pd.DataFrame(disney1.groupby('genre')['total_gross'].mean().sort_values(ascending=True)).astype('int64')
disney7 = disney7.reset_index()
disney7

genre  total_gross
0  Concert/Performance     65281781
1              Western     78294423
2                Drama    102004295
3               Comedy    111116184
4      Romantic Comedy    120102820
5    Thriller/Suspense    140103724
6              Musical    141266983
7               Action    209918484
8            Adventure    223600118

As the above 2 tables reflect data about various genre types and which genre type movies in a dataset has earned how much total gross revenue and what is the mean of those movies. 
The 1st table reflects all the movies category which has performed below and also includes those movies which has performed better than the average and in the second table it includes only those genre movies which has performed better than the average and 3 genre movies type were not better than the average as well.

In [42]:
disney8 = (
    alt.Chart(disney6,width = 500, height = 300)
    .mark_bar().encode(
        x = alt.X("genre:O",title = "Movies Type",sort="-y"),
        y=alt.Y("total_gross:Q", title="Gross Revenue"),
    )
    .properties(title = "Revenue by Movies Type")
)
disney8

alt.Chart(...)

In [43]:
disney9 = (
    alt.Chart(disney7,width = 500, height = 300)
    .mark_bar().encode(
        x = alt.X("genre:O",title = "Movies Type",sort="-y"),
        y=alt.Y("total_gross:Q", title="Gross Revenue"),
    )
    .properties(title = "Revenue by Movies Type")
)
disney9

alt.Chart(...)

As the above 2 tables reflect data about various genre types and which genre type movies had highest total gross revenue based on the mean which was calculated above whether the movie genre was above the benchmark or the below the benchmark and compare which category of movies has highest gross revenue.
As the first three categories movies pattern was same as Adventure, Action, and was highest gross revenue whether it was above the benchmark or including the benchmark that affected the outcome whereas Western and Concert/ Performance were at the eighth and ninth position respectively based on the gross revenue it clearly states that as it was movie    

In [44]:
disney10 = disney_df["release_date"].dt.year 
disney10

564    2015
524    2012
578    2016
571    2016
558    2015
       ... 
511    2011
20     1977
29     1981
355    2002
27     1981
Name: release_date, Length: 579, dtype: int64

In [45]:
disney11 = disney10.value_counts()
disney11

1995    32
1994    30
1996    28
1993    27
1997    23
2002    22
1998    22
1992    22
1999    21
2006    19
2003    19
2000    19
2004    19
2005    17
1991    16
2009    16
1990    15
2011    14
2010    14
2007    14
2016    14
2001    14
2008    13
1988    12
2014    12
1989    11
2013    11
2015    11
1987    10
2012    10
1986     7
1985     6
1983     4
1981     4
1977     4
1982     3
1961     3
1940     2
1984     2
1970     2
1980     2
1971     1
1968     1
1963     1
1959     1
1967     1
1954     1
1975     1
1979     1
1946     1
1950     1
1955     1
1937     1
1962     1
Name: release_date, dtype: int64

As the above 2 tables I had converted release_date into year and than later on found how many movies were released in every year to find out the trend whether there is an increase in movies release or not. 

In [46]:
disney12 = pd.DataFrame({"year": disney11.index,"count":disney11.values})
disney12

year  count
0   1995     32
1   1994     30
2   1996     28
3   1993     27
4   1997     23
5   2002     22
6   1998     22
7   1992     22
8   1999     21
9   2006     19
10  2003     19
11  2000     19
12  2004     19
13  2005     17
14  1991     16
15  2009     16
16  1990     15
17  2011     14
18  2010     14
19  2007     14
20  2016     14
21  2001     14
22  2008     13
23  1988     12
24  2014     12
25  1989     11
26  2013     11
27  2015     11
28  1987     10
29  2012     10
30  1986      7
31  1985      6
32  1983      4
33  1981      4
34  1977      4
35  1982      3
36  1961      3
37  1940      2
38  1984      2
39  1970      2
40  1980      2
41  1971      1
42  1968      1
43  1963      1
44  1959      1
45  1967      1
46  1954      1
47  1975      1
48  1979      1
49  1946      1
50  1950      1
51  1955      1
52  1937      1
53  1962      1

As wanted to plot in a graph wanted to have data in a tabular format to see how many Disney movies were released in every year. 

In [47]:
disney13 = (
    alt.Chart(disney12,width = 500, height = 300)
    .mark_bar().encode(
        x = alt.X("year:O",title = " Year of Disney Movies"),
        y=alt.Y("count:Q", title=" Total No. of Movies Release In a Year"),
    )
    .properties(title = "Movies Released year-wise")
)
disney13

alt.Chart(...)

In the above chart it clearly states that disney movies has increased especially from 1985 year onwards amd in 1995 it was the highest movies release but after 1995 there was a drop in disney movies release but it was releasing a sufficent movies on yearly basis. Also some of the highest grossing movies were released especially after 2010 which has been noticed where we had done our analysis above in disney_df

In [48]:
disney14 = disney_df
disney14

movie_title release_date      genre  \
564    Star Wars Ep. VII: The Force Awakens   2015-12-18  Adventure   
524                            The Avengers   2012-05-04     Action   
578            Rogue One: A Star Wars Story   2016-12-16  Adventure   
571                            Finding Dory   2016-06-17  Adventure   
558                 Avengers: Age of Ultron   2015-05-01     Action   
..                                      ...          ...        ...   
511                                Zokkomon   2011-04-22  Adventure   
20   The Many Adventures of Winnie the Pooh   1977-03-11        NaN   
29                                Condorman   1981-08-07     Action   
355                     Frank McKlusky C.I.   2002-01-01        NaN   
27                                      Amy   1981-03-20      Drama   

     total_gross  inflation_adjusted_gross  
564    936662225                 936662225  
524    623279547                 660081224  
578    529483936                 529483936  
571    486295561                 486295561  
558    459005868                 459005868  
..           ...                       ...  
511         2815                      2984  
20             0                         0  
29             0                         0  
355            0                         0  
27             0                         0  

[579 rows x 5 columns]

In [49]:
disney14["year"] = disney14["release_date"].dt.year
disney14

movie_title release_date      genre  \
564    Star Wars Ep. VII: The Force Awakens   2015-12-18  Adventure   
524                            The Avengers   2012-05-04     Action   
578            Rogue One: A Star Wars Story   2016-12-16  Adventure   
571                            Finding Dory   2016-06-17  Adventure   
558                 Avengers: Age of Ultron   2015-05-01     Action   
..                                      ...          ...        ...   
511                                Zokkomon   2011-04-22  Adventure   
20   The Many Adventures of Winnie the Pooh   1977-03-11        NaN   
29                                Condorman   1981-08-07     Action   
355                     Frank McKlusky C.I.   2002-01-01        NaN   
27                                      Amy   1981-03-20      Drama   

     total_gross  inflation_adjusted_gross  year  
564    936662225                 936662225  2015  
524    623279547                 660081224  2012  
578    529483936                 529483936  2016  
571    486295561                 486295561  2016  
558    459005868                 459005868  2015  
..           ...                       ...   ...  
511         2815                      2984  2011  
20             0                         0  1977  
29             0                         0  1981  
355            0                         0  2002  
27             0                         0  1981  

[579 rows x 6 columns]

As wanted to evaluate in which year disney movies has earned highest gross revenue based on year as the data was in the yyyy-mm-dd format so wanted to convert in a year to find out how much total revenue was generated in each year 

In [50]:
disney15 = pd.DataFrame(disney14.groupby('year')['total_gross'].sum()).astype('int64')
disney15 = disney15.reset_index()
disney15

year  total_gross
0   1937    184925485
1   1940    167620000
2   1946     65000000
3   1950     85000000
4   1954     28200000
5   1955     93600000
6   1959      9464608
7   1961    188599723
8   1962      9230769
9   1963     22182353
10  1967    141843000
11  1968     21540050
12  1970     74282749
13  1971     17871174
14  1975     31916500
15  1977    102717599
16  1979     35841901
17  1980     13900000
18  1981     59899231
19  1982     38617984
20  1983     38833732
21  1984     70999495
22  1985     72981626
23  1986    254523258
24  1987    560208813
25  1988    479129298
26  1989    525447310
27  1990    613857056
28  1991    647050340
29  1992    854534354
30  1993    701652109
31  1994   1092255030
32  1995   1131964294
33  1996   1134044721
34  1997    803617066
35  1998   1229279167
36  1999   1200292143
37  2000   1098929717
38  2001    881246784
39  2002   1159269061
40  2003   1564114393
41  2004   1157899879
42  2005    970646842
43  2006   1427356974
44  2007   1436787754
45  2008    968276109
46  2009   1215142753
47  2010   1518975880
48  2011   1103207068
49  2012   1452972057
50  2013   1821352070
51  2014   1514179473
52  2015   2495662696
53  2016   2873393105

As wanted to graph in which year how much revenue was collected by Disney movies in each year so sum up all the gross revenue for the respective year

In [51]:
disney16 = (
    alt.Chart(disney15,width = 500, height = 300)
    .mark_bar().encode(
        x = alt.X("year:O",title = "Revenue Year Wise",sort="-y"),
        y=alt.Y("total_gross:Q",title="Sum of Gross Revenue of all movies released in a year"),
    )
    .properties(title = "Sum of Revenue based on year wise")
)
disney16

alt.Chart(...)

As in the above graph shows that even though in the previous chart we saw that number of movies release has decreased but substancial movies were released and the revenue was highest in the year 2016 even though no. of movies was not highest. Also the trend of highest revenue is in the decending order from 2016 for majorly for most of the years and 2016 being the highest revenue earned during that year.    

# References

Not all the work in this notebook is original. Some parts were borrowed from online resources

## Resources

* [Codes for Python](https://stackoverflow.com/)
    * The codes used in this project was taken from **Stackoverflow**
* [Codes for Python](https://pandas.pydata.org)
    * The codes used in this project was taken from **Pandas Pydata**
* [Codes for Python](https://www.kaggle.com/discussion)
    * The codes used in this project was taken from **Kaggle**